### PR TITLE
Ignore case on handlerVersion

### PR DESCRIPTION
**Pull request by jkotalik**
For #921 . Also refactored tests a bit.

### CODE REVIEW

**Comment by pakrym** [comment]
Add an empty name/value test cases to make sure it doesn't crash.

File: test/CommonLibTests/ConfigUtilityTests.cpp [hunk 1]
```diff
@@ -8,68 +8,50 @@ using ::testing::NiceMock;
...
+                .WillOnce(DoAll(testing::SetArgPointee<0>(SysAllocString(key.c_str())), testing::Return(S_OK)))
+                .WillOnce(DoAll(testing::SetArgPointee<0>(SysAllocString(L"value")), testing::Return(S_OK)));
+
+            HRESULT hr = ConfigUtility::FindHandlerVersion(element.get(), &handlerVersion);
+
+            EXPECT_STREQ(handlerVersion.QueryStr(), expected.c_str());
+        }
+    };
+
+    TEST_F(ConfigUtilityTest, WriteToFileCheckContentsWritten)
```
[the remark this answers is not in the document]
Comment (jkotalik) [reply test/CommonLibTests/ConfigUtilityTests.cpp]: copy pasta

**Comment by jkotalik** [comment]
@dotnet-bot test this please

**Comment by jkotalik** [comment]
In the appveyor build there looks to be a double free with strings (error code is Heap corruption). We may not be able to free the bstrs. Will try to repro.

**Comment by jkotalik** [comment]
🆙 📅

**Pull request merged**